ddtrace/internal/datadog/profiling/dd_wrapper/include/sample.hpp

```diff
@@ -15,6 +15,34 @@ extern "C"
 
 namespace Datadog {
 
+namespace internal {
+
+struct StringArena
+{
+    // TODO: pick a reasonable default size and explain it. Goal is to be big
+    // enough to accommodate "average" sample string data without need for much
+    // growth (new allocs), but without too much wasted space in the typical case.
+    // User-provided things like function and task names can be arbitrarily-sized.
+    // Basically want (# of frames) * (name + filename size) + (# of labels) * (reasonable label size)
+    // where the sizes are a reasonable expected value.
+    static constexpr size_t DEFAULT_SIZE = 64 * 1024;
```

Comment:
is there a way to look at ingested profiles (sent from python services)? it would be handy to have a statistics of strings from those and set the default based on that. 

Oh wait, this is per sample not per profile. Then, is instrumentation telemetry the only way?

Reply:
Yeah, good idea. I think we can compute how big this would be from a sample in a profile, since we'd have all the labels and frames. I'll collect some data and update this based on what I find 👍

Reply:
I downloaded a bunch of Python relenv profiles and computed the total size of strings associated with each sample (label values and function/file names, basically). The largest I saw for a single sample was ~10000 bytes, but I think that might have been a native profile that my query incorrectly returned. Most of them were actually python profiles, though, so I think the numbers are okay. So I think 16KiB would be a reasonable size to let use usually only have one "chunk" and not need to do more allocations.